src/Sarif/Writers/PrereleaseCompatibilityTransformer.cs

```diff
@@ -1304,6 +1304,10 @@ private static bool ApplyChangesFromTC31(JObject sarifLog)
 
                             // https://github.com/oasis-tcs/sarif-spec/issues/317
                             SetResultKindAndFailureLevel(result);
+
+                            // Bug fix: threadFlowLocation.kind getting lost. https://github.com/microsoft/sarif-sdk/issues/1502
+                            // "kind" should be transformed to "kinds".
+                            ConvertThreadFlowLocationKindToKinds(result);
```

Comment:
why i added this transformation here? I noticed that kind -> kinds changed is first seen in [http://json.schemastore.org/sarif-2.0.0-csd.2.beta.2019-01-24](http://json.schemastore.org/sarif-2.0.0-csd.2.beta.2019-01-24)

the previous one which i am basing on the switch statement in this file should be [http://json.schemastore.org/sarif-2.0.0-csd.2.beta.2019-01-09](http://json.schemastore.org/sarif-2.0.0-csd.2.beta.2019-01-09) has "threadflowlocation.kind". So my best guess is, that this change was introduced in TC31. Although, i couldnt find any issue (open or closed) in oasis-spec github issue list.

Please validate this from memory or if there is a way to ascertain when thsi was introduced.

thanks :)

Reply:
Per our discussion, this is as good a guess as any of where this belongs, given the complicated history of this issue.

---
In reply to: [288371386](https://github.com/microsoft/sarif-sdk/pull/1510#discussion_r288371386) [](ancestors = 288371386)

Reply:
thanks for confirming!

---
In reply to: [288701803](https://github.com/microsoft/sarif-sdk/pull/1510#discussion_r288701803) [](ancestors = 288701803,288371386)